pypeit/edgetrace.py

```diff
@@ -3970,326 +3988,235 @@ def fully_masked_traces(self, flag=None, exclude=None):
 
-        # Check that there are traces to refine!
+        # Check that there are traces to match!
         if self.is_empty:
-            msgs.error('No traces to refine.')
+            msgs.error('No traces to match.')
 
         # The PCA decomposition must have already been determined
         if self.pcatype is None:
             msgs.error('Must first run the PCA analysis for the traces; run build_pca.')
 
-        # Get the file to use when parsing the mask design information
-        _design_file = (None if self.traceimg.files is None else self.traceimg.files[0]) \
-                            if design_file is None else design_file
-        if _design_file is None or not os.path.isfile(_design_file):
-            msgs.error('Slit-mask design file not found or none provided.')
-
-        # Get the paramters to use
-        maxiter = self.par['mask_reg_maxiter']
-        maxsep = self.par['mask_reg_maxsep']
-        sigma = self.par['mask_reg_sigrej']
-        ignore_alignment = self.par['ignore_alignment']
-
-        # TODO: Set allow_resync and design_file to be a parameters, as
-        # well?
-
-        # Read the design data
-        msgs.info('Reading slit-mask design information from: {0}'.format(_design_file))
-        if self.spectrograph.get_slitmask(_design_file) is None:
-            msgs.error('Unable to read design file or no slit-mask design reader '
-                       'defined for {0}.'.format(self.spectrograph.spectrograph))
-
-        # Match both left and right edges simultaneously
-        x_design = np.array([self.spectrograph.slitmask.bottom[:,0],
-                             self.spectrograph.slitmask.top[:,0]]).T.ravel()
-        reference_row = self.left_pca.reference_row if self.par['left_right_pca'] \
-                            else self.pca.reference_row
-        x_det = self.edge_fit[reference_row,:]
-
+        # `traceimg` must have knowledge of the flat frame that built it
+        if self.spectrograph.get_slitmask(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read slitmask design info')
+        if self.spectrograph.get_grating(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read grating info')
+        if self.spectrograph.get_amapbmap(self.traceimg.files[0]) is None:
+            msgs.error('Unable to read amap and bmap')
+
+        # Match left and right edges separately
+        # Position of the slits in mm from the slit-mask design
+        xcen_mm = self.spectrograph.slitmask.center[:, 0]
+        sortindx = np.argsort(xcen_mm)
+
+        xtop_mm = self.spectrograph.slitmask.top[:, 0]
+        xbot_mm = self.spectrograph.slitmask.bottom[:, 0]
+
+        ytop_mm = self.spectrograph.slitmask.top[:, 1]
+        ybot_mm = self.spectrograph.slitmask.bottom[:, 1]
+
+        slitindx = self.spectrograph.slitmask.slitindx
+        slit_id = self.spectrograph.slitmask.slitid
+
+        npoints = 250
+        nslits = self.spectrograph.slitmask.nslits
```

Comment:
Why are you assuming the wavelengths are fixed here, I'm lost?

Reply:
This is hard-coded for the DEIMOS optical model. I moved this part to `mask_to_pixel_coordinates()`